libs/model/src/snapshot/createSnapshotProposal.command.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EventNames,
+  InvalidInput,
+  InvalidState,
+  logger,
+  type Command,
+} from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import fetch from 'node-fetch';
+import { Op } from 'sequelize';
+import { models } from '../database';
+import { emitEvent } from '../utils/utils';
+
+const log = logger(import.meta);
+
+const Errors = {
+  FailedParsing: 'Failed to parse id and/or event from payload',
+  SpaceNotDefined: 'Space is not defined',
+  ProposalNotFound: 'Proposal not found',
+};
+
+export async function fetchNewSnapshotProposal(id: string) {
+  try {
+    const response = await fetch('https://hub.snapshot.org/graphql', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify({
+        query: `
+         query($id: String!) {
+            proposal(id: $id) {
+            id
+            title
+            body
+            start
+            choices
+            end
+            space {
+              name
+              id
+            }
+          }
+        }`,
+        variables: { id },
+      }),
+    });
+
+    const proposal = await response.json();
+    proposal.expire = proposal.end;
+
+    return proposal;
+  } catch (err) {
+    log.error(
+      'Error fetching snapshot proposal from GraphQL endpoint',
+      err as Error,
+    );
+    return err;
+  }
+}
+
+// This should technically be an 'event' but the express adapter
+// does not support events yet
+export function CreateSnapshotProposal(): Command<
```

Comment:
Is this actually creating a snapshot proposal or just enqueuing the event in the outbox for later processing? Feels like an adapter (ACLish) instead of a core domain command.

Reply:
This is indeed not a 'true' command since it just enqueues an event to the outbox hence the comment above the command.